doc/api/child_process.markdown

```diff
@@ -1,359 +1,67 @@
-      }
+    ls.stderr.on('data', (data) => {
+      console.log(`stderr: ${data}`);
     });
 
-Note that once a single socket has been sent to a child the parent can no
-longer keep track of when the socket is destroyed. To indicate this condition
-the `.connections` property becomes `null`.
-It is also recommended not to use `.maxConnections` in this condition.
-
-### child.stderr
-
-* {Stream object}
-
-A `Readable Stream` that represents the child process's `stderr`.
-
-If the child was not spawned with `stdio[2]` set to `'pipe'`, then this will
-not be set.
-
-`child.stderr` is shorthand for `child.stdio[2]`. Both properties will refer
-to the same object, or null.
-
-### child.stdin
-
-* {Stream object}
-
-A `Writable Stream` that represents the child process's `stdin`.
-If the child is waiting to read all its input, it will not continue until this
-stream has been closed via `end()`.
-
-If the child was not spawned with `stdio[0]` set to `'pipe'`, then this will
-not be set.
-
-`child.stdin` is shorthand for `child.stdio[0]`. Both properties will refer
-to the same object, or null.
-
-### child.stdio
-
-* {Array}
-
-A sparse array of pipes to the child process, corresponding with positions in
-the [`stdio`][] option to [`spawn()`][] that have been set to `'pipe'`.
-Note that streams 0-2 are also available as ChildProcess.stdin,
-ChildProcess.stdout, and ChildProcess.stderr, respectively.
-
-In the following example, only the child's fd `1` is setup as a pipe, so only
-the parent's `child.stdio[1]` is a stream, all other values in the array are
-`null`.
-
-    const assert = require('assert');
-    const fs = require('fs');
-    const child_process = require('child_process');
-
-    const child = child_process.spawn('ls', {
-        stdio: [
-          0, // use parents stdin for child
-          'pipe', // pipe child's stdout to parent
-          fs.openSync('err.out', 'w') // direct child's stderr to a file
-        ]
+    ls.on('close', (code) => {
+      console.log(`child process exited with code ${code}`);
     });
 
-    assert.equal(child.stdio[0], null);
-    assert.equal(child.stdio[0], child.stdin);
-
-    assert(child.stdout);
-    assert.equal(child.stdio[1], child.stdout);
-
-    assert.equal(child.stdio[2], null);
-    assert.equal(child.stdio[2], child.stderr);
+Pipes for `stdin`, `stdout` and `stderr` are established between the parent
+Node.js process and the created child. It is possible to stream data through
+these pipes in a non-blocking way. *Note, however, that some programs use
+line-buffered I/O internally. While that does not affect Node.js, it can mean
+that data sent to the child process may not be immediately consumed.*
+
+For convenience, the `child_process` module provides a handful of synchronous
+and asynchronous alternatives to `child_process.spawn()`, each of which are
+documented fully [below][]:
+
+  * `child_process.exec()`: spawns a shell and runs a command within that shell.
```

Comment:
... "returning the stdout and stderr when the command completes". <--- that last is the very important distinction between the exec* and the spawn/fork.